src/include/sof/schedule.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2019, Intel Corporation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of the Intel Corporation nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Bartosz Kokoszko <bartoszx.kokoszko@linux.intel.com>
+ */
+
+/* Generic schedule api header */
+#ifndef __INCLUDE_SOF_SCHEDULER_H__
+#define __INCLUDE_SOF_SCHEDULER_H__
+
+#include <stdint.h>
+#include <sof/edf_schedule.h>
+
+#define SOF_EDF_TASK		0
+#define SOF_LOW_LATENCY_TASK	1
+
+#define SOF_TASK_LOW	0
+#define SOF_TASK_MED	1
+#define SOF_TASK_HIGH	2
+
+extern uint32_t task_priority[2][3];
+
+struct sof_task {
+	uint16_t core;
+	uint32_t type;	/* edf/ll */
+	uint64_t start;	/* start/timeout */
+	uint32_t priority;
+	void *data;
+	uint64_t (*func)(void *data, uint64_t udelay);
+
+	uint64_t max_rtime;		/* max time taken to run */
+	completion_t complete;
+
+	void *private;
+};
+
+static inline void sof_task_init(struct sof_task *task, uint32_t type,
+	uint32_t priority, uint64_t (*func)(void *data, uint64_t udelay),
+	void *data, uint16_t core)
+
+{
+	task->type = type;
+	task->priority = task_priority[type][priority];
+
+	task->func = func;
+	task->data = data;
+
+	switch (task->type) {
+	case SOF_EDF_TASK:
+		task->core = core;
+		break;
+	case SOF_LOW_LATENCY_TASK:
+		task->core = 0; /* does not matter*/
```

Comment:
This is restrictive, we must allow any core.

Reply:
Done